src/mug.gleam

```diff
@@ -1,20 +1,88 @@
 import gleam/bytes_tree.{type BytesTree}
 import gleam/dynamic.{type Dynamic}
+import gleam/erlang
 import gleam/erlang/atom
 import gleam/erlang/charlist.{type Charlist}
 import gleam/erlang/process
+import gleam/io
+import gleam/list
+import gleam/option.{type Option, None, Some}
+import gleam/result
+import gleam/string
+import mug/internal/ssl_options.{
+  type SslOptionName, Cacertfile, Cacerts, CertsKeys, Verify,
+}
+
+pub type TcpSocket
 
-pub type Socket
+pub type SslSocket
+
+pub type Socket {
+  TcpSocket(TcpSocket)
+  SslSocket(SslSocket)
+}
 
 type DoNotLeak
 
+/// Error occured during a TLS operation.
+///
+/// Reference: https://www.erlang.org/doc/apps/ssl/ssl.html#t:tls_alert/0
+pub type TlsAlert {
+  CloseNotify
+  UnexpectedMessage
+  BadRecordMac
+  RecordOverflow
+  HandshakeFailure
+  BadCertificate
+  UnsupportedCertificate
+  CertificateRevoked
+  CertificateExpired
+  CertificateUnknown
+  IllegalParameter
+  UnknownCa
+  AccessDenied
+  DecodeError
+  DecryptError
+  ExportRestriction
+  ProtocolVersion
+  InsufficientSecurity
+  InternalError
+  InappropriateFallback
+  UserCanceled
+  NoRenegotiation
+  UnsupportedExtension
+  CertificateUnobtainable
+  UnrecognizedName
+  BadCertificateStatusResponse
+  BadCertificateHashValue
+  UnknownPskIdentity
+  NoApplicationProtocol
+}
+
 /// Errors that can occur when working with TCP sockets.
 ///
 /// For more information on these errors see the Erlang documentation:
 /// - https://www.erlang.org/doc/man/file#type-posix
 /// - https://www.erlang.org/doc/man/inet#type-posix
 ///
 pub type Error {
+  /// The `ssl` application was not started/depended upon.
+  /// Either call `tls.start()` before calling any TLS methods, or (better yet) depend on the `ssl`
+  /// erlang application in [`gleam.toml`](https://gleam.run/writing-gleam/gleam-toml/).
+  SslNotStarted
```

Comment:
Remove this please. It should never happen and the documentation gives incorrect advice.

Reply:
I had written this code before reading up on OTP behaviours 😅